angular-project/src/app/card-tracker/card-tracker.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, Input, OnInit } from '@angular/core';
+import TimeData from 'src/assets/json/data.json';
+import { TimeInterface } from 'src/app/time-data-interface';
+
+@Component({
+  selector: 'app-card-tracker',
+  templateUrl: './card-tracker.component.html',
+  styleUrls: ['./card-tracker.component.scss']
+})
+export class CardTrackerComponent implements OnInit {
+  @Input() currentPeriod: string = 'daily' // initial value 'daily
+  @Input() currentActivity: TimeInterface = TimeData[0]; // initial value TimeData[0]
+  // get hours for selected time period
+  getHours (period:string) {
+    if (period === 'daily')
+      return this.currentActivity.timeframes.daily;
+    else if (period === 'weekly')
+      return this.currentActivity.timeframes.weekly;
+    else // (period === 'monthly')
+      return this.currentActivity.timeframes.monthly;
+  }
```

Comment:
Так строка `period` совпадает с названием свойства в timeframes, то можно сделать автоматический вызов по нему
https://learn.javascript.ru/object#kvadratnye-skobki
```suggestion
  getHours (period:string) {
    returh this.currentActivity.timeframes[period];
  }
```